src/ContentRepository/ServicesComponent.cs

```diff
@@ -1550,6 +1550,51 @@ void SetSettingsDescription(string name, string description)
 
         #endregion
 
+        private void Patch_PreviewSwitches(PatchExecutionContext context)
+        {
+            var logger = context.GetService<ILogger<ServicesComponent>>();
+
+            #region String resource changes
+
+            var rb = new ResourceBuilder();
+            rb.Content("CtdResourcesEF.xml")
+                .Class("Ctd-Folder")
+                .Culture("en")
+                .AddResource("PreviewEnabled-DisplayName", "Preview enabled")
+                .AddResource("PreviewEnabled-Description", "Switch of the preview generation on this subtree. Enables, disables or gets from parent.")
+                .Culture("hu")
+                .AddResource("PreviewEnabled-DisplayName", "Előnézet engedélyezése")
+                .AddResource("PreviewEnabled-Description", "Az előnézeti képek generálásának kapcsolója ezen a részfán. Be- vagy kikapocsolja, vagy a szülőről veszi.");
+            rb.Apply();
+
+            #endregion
+
+            #region CTD changes
+
+            try
+            {
+                var cb = new ContentTypeBuilder(context.GetService<ILogger<ContentTypeBuilder>>());
+                cb.Type("Folder")
+                    .Field("PreviewEnabled", "Choice")
+                    .DisplayName("$Ctd-Folder,PreviewEnabled-DisplayName")
+                    .Description("$Ctd-Folder,PreviewEnabled-Description")
+                    .VisibleBrowse(FieldVisibility.Hide)
+                    .VisibleEdit(FieldVisibility.Advanced)
+                    .VisibleNew(FieldVisibility.Advanced)
+                    .Configure("AllowMultiple", "false")
+                    .Configure("AllowExtraValue", "false")
+                    .Configure("Options", "<Enum type=\"SenseNet.ContentRepository.PreviewEnabled\"/>");
+                cb.Apply();
+            }
+            catch (Exception ex)
+            {
+                logger.LogWarning(ex, "Error during CTD changes.");
+            }
+
+            #endregion
```

Comment:
To avoid a huge change, switch ON preview generation on the Root at the end of the patch.